src/admin/BulkUserCreationPage.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { FC } from 'react';
+import { Box, Label, Heading, Container, Button, Text, Textarea, Alert, Select } from 'theme-ui';
+import { useFormik } from 'formik';
+import * as yup from 'yup';
+
+import { Role, CreateUserCommand } from '../api';
+import useBulkUserCreation from './hooks/useBulkUserCreation';
+import useRoles from './hooks/useRoles';
+
+const AnyBox = Box as any;
+
+const BulkUserCreationPage: FC = () => {
+  const {
+    create,
+    loading: creatingUsers,
+    error: errorCreatingUsers,
+    createdUsers,
+  } = useBulkUserCreation();
+  const { roles, loading: loadingRoles, error: errorLoadingRoles } = useRoles();
+
+  const form = useFormik<{ emailsString: string; role: Role['name'] }>({
+    initialValues: {
+      role: roles[0]?.name || '',
+      emailsString: '',
+    },
+    enableReinitialize: true,
+    validationSchema: yup.object().shape({
+      role: yup.string().required(),
+      emailsString: yup.string().required(),
+    }),
+    onSubmit: ({ role, emailsString }) => {
+      const emails = getEmails(emailsString);
+      const command: CreateUserCommand[] = emails.map((email) => ({ email, roles: [role] }));
+
+      if (
+        window.confirm(
```

Comment:
nice

Reply:
Seemed dangerous, thought I'd throw it in.